### ISSUE TITLE
Sample applications throws on dns chalanges

### ISSUE THREAD

**Issue by dv00d00** [ffMathy/FluffySpoon.AspNet.LetsEncrypt#23]
First of all, thx for the effort of bringing LetsEncrypt goodness to .net devs!
I am trying to deploy asp.net core app with lets encrypt certificates and decided to try this library.
Unfortunately, I've spent a whole day fighting lets encrypt issues yesterday.

I believe I found a bug/issue in the [LetsEncryptRenewalService.cs](https://github.com/ffMathy/FluffySpoon.AspNet.LetsEncrypt/blob/master/src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs#L231)

With this line present, I am getting this execution log: 

```bash
Hosting environment: Development
Content root path: C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt.Sample
Now listening on: http://[::]:9999
Now listening on: https://[::]:19999
Application started. Press Ctrl+C to shut down.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Checking to see if in-memory LetsEncrypt certificate needs renewal.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Checking to see if existing LetsEncrypt certificate has been persisted and is valid.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Ordering LetsEncrypt certificate for domains ffmathyletsencrypt2.ngrok.io.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Validating all pending order authorizations.
warn: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Exception occured renewing certificates: 'Fail to load resource from 'https://acme-staging-v02.api.letsencrypt.org/acme/chall-v3/26169143/dWaKCw'.
urn:ietf:params:acme:error:malformed: No such challenge.'
Certes.AcmeRequestException: Fail to load resource from 'https://acme-staging-v02.api.letsencrypt.org/acme/chall-v3/26169143/dWaKCw'.
urn:ietf:params:acme:error:malformed: No such challenge
   at Certes.Acme.IAcmeHttpClientExtensions.Post[T](IAcmeHttpClient client, Uri uri, Object payload, Boolean ensureSuccessStatusCode)
   at Certes.Acme.EntityContext`1.Resource()
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateChallengesAsync(IEnumerable`1 challengeContexts) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 287
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 251
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 273
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.AcquireNewCertificateForDomains(String[] domains) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 159
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 139
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 146
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceWithErrorHandlingAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 105
```

with it commented I am getting this
```bash
Hosting environment: Development
Content root path: C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt.Sample
Now listening on: http://[::]:9999
Now listening on: https://[::]:19999
Application started. Press Ctrl+C to shut down.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Checking to see if in-memory LetsEncrypt certificate needs renewal.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Checking to see if existing LetsEncrypt certificate has been persisted and is valid.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Ordering LetsEncrypt certificate for domains ffmathyletsencrypt2.ngrok.io.
info: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Validating all pending order authorizations.
warn: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Exception occured renewing certificates: 'One or more LetsEncrypt orders were invalid. Make sure that LetsEncrypt can contact the domain you are trying to request an SSL certificate for, in order to verify it..'
FluffySpoon.AspNet.LetsEncrypt.Exceptions.OrderInvalidException: One or more LetsEncrypt orders were invalid. Make sure that LetsEncrypt can contact the domain you are trying to request an SSL certificate for, in order to verify it. ---> System.AggregateException: One or more errors occurred. (urn:ietf:params:acme:error:unauthorized: Invalid response from http://ffmathyletsencrypt2.ngrok.io/.well-known/acme-challenge/dE2H582T8rOFyEfP__0D7Y1oOSBgnNHVJZOn_t1CXac [2600:1f16:d83:1200:6510:cd35:dbb5:e85d]: 404 (challenge type http-01)) ---> System.Exception: urn:ietf:params:acme:error:unauthorized: Invalid response from http://ffmathyletsencrypt2.ngrok.io/.well-known/acme-challenge/dE2H582T8rOFyEfP__0D7Y1oOSBgnNHVJZOn_t1CXac [2600:1f16:d83:1200:6510:cd35:dbb5:e85d]: 404 (challenge type http-01)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 269
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 273
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.AcquireNewCertificateForDomains(String[] domains) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 159
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 139
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 146
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceWithErrorHandlingAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 105

```

It looks like the first log captures exception from the `Certes` itself, while the second demonstrates correct behavior.

Please let me know if I can configure something inside my code to make it work for http challenges?

If anything I would be glad to help to resolve the issue.  (adding tests would be helpful)

**Comment by ffMathy** [comment]
So sorry to hear your day was wasted. Have you tried the version that was released today?

**Comment by dv00d00** [comment]
I've checked out master after the change so I believe this is still the case.

**Comment by dv00d00** [comment]
tldr: If I am commenting out 
```csharp
challengeContextTasks.AddRange(allAuthorizations.Select(x => x.Dns()));
```
from LetsEncryptRenewalService.cs I am starting getting OrderInvalidException (acme client got chalange and stored it)

For some reason a call to .Dns() challenge throws an `Certes.AcmeRequestException`.

I am still not sure if it is an issue tbh

**Comment by ffMathy** [comment]
Aha! I'll fix it.

**Comment by ffMathy** [comment]
What does the exception contain?

**Comment by dv00d00** [comment]
```
warn: FluffySpoon.AspNet.LetsEncrypt.ILetsEncryptRenewalService[0]
      Exception occured renewing certificates: 'Fail to load resource from 'https://acme-staging-v02.api.letsencrypt.org/acme/chall-v3/26169143/dWaKCw'.
urn:ietf:params:acme:error:malformed: No such challenge.'
Certes.AcmeRequestException: Fail to load resource from 'https://acme-staging-v02.api.letsencrypt.org/acme/chall-v3/26169143/dWaKCw'.
urn:ietf:params:acme:error:malformed: No such challenge
   at Certes.Acme.IAcmeHttpClientExtensions.Post[T](IAcmeHttpClient client, Uri uri, Object payload, Boolean ensureSuccessStatusCode)
   at Certes.Acme.EntityContext`1.Resource()
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateChallengesAsync(IEnumerable`1 challengeContexts) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 287
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 251
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.ValidateOrderAsync(String[] domains, IOrderContext order) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 273
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.AcquireNewCertificateForDomains(String[] domains) in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 159
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 139
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 146
   at FluffySpoon.AspNet.LetsEncrypt.LetsEncryptRenewalService.RunOnceWithErrorHandlingAsync() in C:\work\FluffySpoon.AspNet.LetsEncrypt\src\FluffySpoon.AspNet.LetsEncrypt\LetsEncryptRenewalService.cs:line 105
```

**Comment by ffMathy** [comment]
Hmmm can you try updating Certes of the repo?

**Comment by dv00d00** [comment]
Certes is of the latest version atm, but I've pulled your commit, it has fixed the issue.
Amazingly fast response time, thank you so much!

**Comment by ffMathy** [comment]
No worries!

If you like the library, consider sponsoring! Even a dollar counts.
